code/modules/detective_work/forensicmachines.dm

```diff
@@ -0,0 +1,316 @@
+// Designs
+/datum/design/dnaforensics
+	name = "Machine Design (DNA analyzer)"
+	desc = "DNA analyzer for forensic DNA analysis of objects."
+	id = "dnaforensics"
+	req_tech = list("programming" = 2, "combat" = 2, "magnets" = 2)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/dnaforensics
+	category = list ("Misc. Machinery")
+
+/obj/item/circuitboard/dnaforensics
+	name = "circuit board (DNA analyzer)"
+	build_path = /obj/machinery/dnaforensics
+	board_type = "machine"
+	origin_tech = "programming=2;combat=2"
+	req_components = list(
+		/obj/item/stock_parts/micro_laser = 2,
+		/obj/item/stock_parts/manipulator = 1,)
+
+/datum/design/microscope
+	name = "Machine Design (Forensic Microscope)"
+	desc = "Microscope capable of magnifying images 3000 times"
+	id = "microscope"
+	req_tech = list("programming" = 2, "combat" = 2, "magnets" = 2)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/microscope
+	category = list ("Misc. Machinery")
+
+/obj/item/circuitboard/microscope
+	name = "circuit board (Microscope)"
+	build_path = /obj/machinery/microscope
+	board_type = "machine"
+	origin_tech = "programming=2;combat=2"
+	req_components = list(
+		/obj/item/stock_parts/micro_laser = 1,
+		/obj/item/stack/sheet/glass = 1)
+
+// DNA machine
+/obj/machinery/dnaforensics
+	name = "\improper DNA analyzer"
+	desc = "A high-tech machine that is designed to sequence DNA samples."
+	icon = 'icons/obj/forensics/forensics.dmi'
+	icon_state = "dnaopen"
+	layer = BELOW_OBJ_LAYER
+	anchored = TRUE
+	density = TRUE
+
+	var/obj/item/forensics/swab = null
+	var/scanning = FALSE
+	var/report_num = FALSE
+
+/obj/machinery/dnaforensics/Initialize(mapload)
+	. = ..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/dnaforensics(null)
+	component_parts += new /obj/item/stock_parts/micro_laser(null)
+	component_parts += new /obj/item/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stock_parts/micro_laser(null)
+	RefreshParts()
+
+/obj/machinery/dnaforensics/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>You can <b>Alt-Click</b> to eject the current sample. <b>Click while holding a sample</b> to insert a sample. <b>Click with an empty hand</b> to operate.</span>"
+
+/obj/machinery/dnaforensics/attackby(obj/item/W as obj, mob/user as mob)
+
+	if(swab)
+		to_chat(user, "<span class='warning'>There is already a test tube inside the scanner.</span>")
+		return
+
+	if(istype(W, /obj/item/forensics/swab))
+		to_chat(user, "<span class='notice'>You insert [W] into the DNA analyzer.</span>")
+		user.unEquip(W)
+		W.forceMove(src)
+		swab = W
+		update_icon()
+		return
+	..()
+
+/obj/machinery/dnaforensics/attack_hand(mob/user)
+
+	if(!swab)
+		to_chat(user, "<span class='warning'>The scanner is empty!</span>")
+		return
+	scanning = TRUE
+	update_icon()
```

Comment:
you shouldn't be calling `update_icon()` on its own, instead call update_appearance(UPDATE_ICON) (or something like that, I might be slightly off on the actual flag)